pints/_transformation.py

```diff
@@ -0,0 +1,574 @@
+#
+# Transformation functions
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pints
+import numpy as np
+from scipy.special import logit, expit
+
+
+class Transform(object):
+    """
+    Abstract base class for objects that provide some convenience parameter
+    transformation from the model parameter space to a search space.
+
+    If ``t`` is an instance of a ``Transform`` class, you can apply
+    the transformation of a parameter vector from the model space ``p`` to the
+    search space ``x`` by using ``x = t.to_search(p)`` and the inverse by using
+    ``p = t.to_model(x)``.
+    """
+    def apply_log_pdf(self, log_pdf):
+        """
+        Returns a transformed log-PDF class.
+        """
+        return TransformedLogPDF(log_pdf, self)
+
+    def apply_error_measure(self, error_measure):
+        """
+        Returns a transformed error measure class.
+        """
+        return TransformedErrorMeasure(error_measure, self)
+
+    def apply_boundaries(self, boundaries):
+        """
+        Returns a transformed boundaries class.
+        """
+        return TransformedBoundaries(boundaries, self)
+
+    def jacobian(self, x):
+        """
+        Returns the Jacobian for a parameter vector ``x`` in the search space.
```

Comment:
Probably worth adding the maths to say _which_ Jacobian, of the many involved in time series inference :-)

Reply:
Which of the many involved in time series inference?

Reply:
Like, dx/dwhat?

Reply:
Do you want something like "Jacobian of the transformation"?

Reply:
No I just want to docstring to define the jacobian. What are you taking the derivative of, and with respect to what? What are x and y in dx/dy?

Reply:
Ha, OK, well, Jacobian is _the_ matrix :smile:

Reply:
Haha yes "Transformation Jacobian" should probably be clear enough. Although I personally get sometimes confused whether the Jacobian is defined as dx/dy or dy/dx, because in theory both work equally fine. :D

Reply:
Yup, in fact, they just hold a inverse relation, according to the inverse function theorem (awesome!). But yes, we need _one_ of them but not the other(!) -- we need the dg^{-1}/y in your y = g(x) notation! I will add this in the docstring.

Reply:
Updated.